_posts/2023-01-04-trino-2022-the-rabbit-reflects.md

```diff
@@ -0,0 +1,581 @@
+
+2022 was the year of resiliency for Trino. One of the longest requested features
+that our user had was adding a [fault-tolerant mechanism to Trino
+]({{site.url}}/docs/current/admin/fault-tolerant-execution.html) akin to
+query engines like Apache Spark. Users wanted the ability to take the queries
+that they were running in Trino, and scale those queries to larger data or
+resource intensive queries. Experimental features were implemented in late 2021
+for [automatic query retries](https://github.com/trinodb/trino/pull/9361) and
+earlier this year [task-level retries
+](https://github.com/trinodb/trino/pull/9818). The efforts for these features
+were codenamed [Project Tardigrade]({{site.url}}/episodes/32.html).
+
+This feature relies on storing intermediate data between task shuffles to
+persist data in what was called an exchange spool. The first iteration of this
+was AWS S3, but eventually Azure Blob Storage, Google Cloud Storage, and local
+storage were included. As the Project Tardigrade engineers started [improving
+performance and fixing bugs]({% post_url 2022-02-16-tardigrade-project-update %})
+in fault-tolerant execution as they and a few users tested the early
+implementation. Later, memory efficiency for aggregations, faster data
+transfers, and dynamic filtering with fault-tolerant query execution were added.
+The [launch of fault-tolerant execution
+]({% post_url 2022-05-05-tardigrade-launch %}) launch happened at
+Cinco de Trino. The first iterations only applied to queries being run on
+object-storage connectors such as Hive, Iceberg, and Delta Lake. Recently,
+support for MySQL, PostgreSQL, and SQL Server were added. These contributions
+added a foundation for other JDBC connectors.
+
+### SQL language improvements
+
+Here are all the notable SQL features that made it in to Trino this year:
+
+* [`MERGE` statement support]({{site.url}}/docs/current/sql/merge.html) is
+ the most impactful SQL feature released this year. `MERGE` allows users to
+ implement `INSERT`, `UPDATE`, and `DELETE` statements in one fell swoop.
+ `MERGE` is not simply syntax sugar, it actually has profound performance
+ implications as now a lot of your operations can be merged (pun intended) from
+ multiple tasks into a single scan over data. This functionality is absolutely
+ critical for positioning Trino as a data lakehouse query engine. `MERGE` is 
+ currently available in Hive, Iceberg, Delta Lake, Kudu, and Raptor connectors.
+ We discussed this and did a demo with `MERGE` on the recent [Trino Community
+ Broadcast with Iceberg]({{site.url}}/episodes/40.html).
+* Another massive update was the introduction of [polymorphic table functions
+ ]({% post_url 2022-07-22-polymorphic-table-functions %}) ([PTFs
+ ]({{site.url}}/docs/current/functions/table.html)). Table functions
+ initially released with some initial passthrough query functionality that we
+ see in connectors like Pinot, Elasticsearch, MySQL, PostgreSQL, and many
```

Comment:
```suggestion
 see in connectors like Pinot, Elasticsearch, MySQL, PostgreSQL, and
```

Reply:
what about "and other JDBC connectors" and I'll link to the PR that points to all of them